### PR TITLE
V3 remove jsxstyle

### PR DESCRIPTION
**Pull request by ChristopherBiscardi**
Nothing like code to get a [discussion started](https://github.com/threepointone/glamor/pull/231#issuecomment-291631541) :)

This PR removes glamor-jsxstyle and replaces it with suggestions to use @kentcdodds' [glamorous](https://github.com/kentcdodds/glamorous) which supports jsxstyle syntax in addition to object-style styled-components (and is built on glamor).

### CODE REVIEW

**Comment by donaldpipowitch** [comment]
Damn, you were faster than me :D 
https://github.com/threepointone/glamor/issues/83#issuecomment-291765534

Do want to wait with this PR? Or do it before https://github.com/threepointone/glamor/issues/229?

**Comment by ChristopherBiscardi** [comment]
All that's left is to switch to typescript right? Everything else looks fairly done: 

>  ~Remove deprecated APIs in v3 branch. (#231)~
>  Switch to TypeScript.
>  ~Add deprecation warnings in v2.~
>  ~Write a migration guide.~

In that case removing a subpackage seems like less work than rewriting it and then removing it 😂. I'm pretty ambivalent about it though.

**Comment by ChristopherBiscardi** [comment]
oh wait, I jumped the gun on crossing out deprecations and a migration for the helpers yet in v2. whoops. Was thinking about core and such which already have those.

File: packages/glamor-jsxstyle/example/src/jsxstyle.js [hunk 1]
```diff
@@ -1,17 +0,0 @@
-import React from 'react'
-import { View, Row, Column } from 'glamor-jsxstyle'
```
Comment (kentcdodds) [inline packages/glamor-jsxstyle/example/src/jsxstyle.js]: We need to add these components to glamorous. Right now it's just elements which you can style.

**Comment by threepointone** [comment]
are they equivalent? I remember writing code to handle a number of usecases.

**Comment by kentcdodds** [comment]
> are they equivalent?

They're not quite equivalent. There's still some work that needs to be done for Glamorous components to support props like `media`, `hover`, `component`, etc. Should probably enumerate the features in [this issue](https://github.com/paypal/glamorous/issues/25) to keep track of what glamorous still needs to support. The actual pre-baked components will come from [`glamorous-jsxstyle`](https://github.com/paulmolluzzo/glamorous-jsxstyle), but we need glamorous to support the props. Help appreciated!

**Comment by threepointone** [comment]
I vote we keep this around for longer. Let's not delete it just yet.

**Comment by donaldpipowitch** [comment]
@threepointone But then we would need to restore the helpers just for `jsxstyle` or am I wrong? Because of https://github.com/threepointone/glamor/pull/231#issuecomment-291608497.

**Comment by donaldpipowitch** [comment]
What is the state of this? 

@kentcdodds It looks like https://github.com/paypal/glamorous/issues/25 was closed. 
@threepointone Can you explain _why_ you want to keep this around longer? Then we'd need to re-add the removed helpers from https://github.com/threepointone/glamor/pull/231#issuecomment-291608497 (at least _inside_ glamor-jsxstyle). Also I _personally_ haven't seen the _need_ of an API like this in the community recently.

**Comment by jaredpalmer** [comment]
So @threepointone reached out to me, and I agreed to maintain `glamor-jsxstyle` going forward. My company uses it pretty extensively on several projects.

Before I go ahead and make `jaredpalmer/glamor-jsxstyle`, wondering if it make sense to establish a glamor organization (a la styled-components) at this point? Then we could keep core as glamor/glamor and all of the official package extensions (aphrodite, jsxstyle, etc.) and experiments under the same umbrella org? Apologies ahead of time if this has been discussed previously. 

I've never transferred an npm repo, but from my reading, I believe all that needs to be done is for @threepointone to clone the new repo from wherever it ends up and then add me to the package by running `npm adduser jaredpalmer` so I can have publish rights. @donaldpipowitch @kentcdodds Feel free to jump in here with the correct way to do this.

**Comment by threepointone** [comment]
a glamor org sounds good, but I don't want that to hold up the v3 release. I'll leave it up to you folks.

**Comment by jaredpalmer** [comment]
@threepointone Okay, so if that's the case, I will go ahead and make `jaredpalmer/glamor-jsxstyle` and then we can make the org on a rainy day and then just transfer it later on.

**Comment by jaredpalmer** [comment]
@threepointone @donaldpipowitch Should I also port this over to TS now?

**Comment by kentcdodds** [comment]
Excited to see more momentum here! :clap:

**Comment by threepointone** [comment]
sorry I've not been around 😔

**Comment by threepointone** [comment]
@jaredpalmer your call, you the boss lol

**Comment by donaldpipowitch** [comment]
Hi,

I'm fine with creating a GitHub organization (now or later after a stable v3). 😃  Nice to see you helping out with maintaining glamor @jaredpalmer. Do you know _how_ you want to maintain `jaredpalmer/glamor-jsxstyle`? Do you want to restore the removed helpers from https://github.com/threepointone/glamor/pull/231 _inside_ `jaredpalmer/glamor-jsxstyle` or as a standalone package or in a completely different way?

> @donaldpipowitch Should I also port this over to TS now?

If you want to use TS this would be nice! 😃 Have you ever used TS before? It probably can help you with maintaining the package and it makes a great experience for TS users, if a lot of Glamor related packages are written in TS. In the last week @otbe and our intern @mosbahh worked on converting the glamor code base to TS (and made it possible to develop glamor on Windows). They made great progress (can be seen [here](https://github.com/otbe/glamor/commits/mosbah/v3-typescript)) and I think they'll have a PR soon.

**Comment by jaredpalmer** [comment]
@donaldpipowitch I've been using TypeScript for about a month or so on an internal project, so I'm pretty familiar with it. That being said, I haven't written a consumable library with it yet, so not 100% on best way to setup tsconfigs for multiple targets etc. However, it looks like the link to the fork you provided is exactly what I needed (an hour ago lol). Anyways, I figured it would be easier to just port the existing code + most of build/test setup to get started.

I plan to include the helpers in glamor-jsxstyle for now.

**Comment by jaredpalmer** [comment]
@donaldpipowitch @threepointone Ready to review https://github.com/jaredpalmer/glamor-jsxstyle

**Comment by donaldpipowitch** [comment]
@jaredpalmer I think it looks good. Well, if you want to you can make this a PR to the `v3` branch, if you want to (IMHO) and after that I we could give you push rights to this repo (if you want to and if no one else, has any concerns 😀 ).

**Comment by donaldpipowitch** [comment]
@jaredpalmer Do you know if you want to maintain `glamor-jsxstyle` in your own repo (https://github.com/jaredpalmer/glamor-jsxstyle) or do you want to make a PR to our https://github.com/threepointone/glamor repo (`v3` branch)? FYI: `glamor` is now written in TypeScript.

**Comment by jaredpalmer** [comment]
TS is not a problem. I think it will be easier to just submit a PR to v3

**Comment by threepointone** [comment]
closing this for now, rethinking v3 strategy, reaching out to jared separately on twitter.

**Pull request closed without merging**